modules/electrophysiology_browser/php/models/electrophysioevents.class.inc

```diff
@@ -57,7 +78,340 @@ class ElectrophysioEvents
+                );
+
+                $db->update(
+                    'physiological_event_file',
+                    ['LastUpdate' => date("Y-m-d H:i:s")],
+                    ['PhysiologicalFileID' => $this->_physioFileID]
+                );
+            }
+
+            $taskEvent = $db->pselect(
+                'SELECT * FROM physiological_task_event
+            WHERE PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            $extraColumns = $db->pselect(
+                'SELECT opt.*
+                  FROM physiological_task_event_opt AS opt
+                  WHERE opt.PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            return [
+                'instance'      => $taskEvent[0],
+                'extra_columns' => $extraColumns,
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Deletes one event instance
+     *
+     * @param int $physiologicalTaskEventID PhysiologicalTaskEventID
+     *
+     * @return void
+     */
+    function deleteEvent(int $physiologicalTaskEventID): void
+    {
+        $db = \NDB_Factory::singleton()->database();
+
+        $physioFileID = $db->pselectone(
+            'SELECT PhysiologicalFileID
+              FROM physiological_task_event
+              WHERE PhysiologicalTaskEventID=:taskEventID',
+            ['taskEventID' => $physiologicalTaskEventID]
+        );
+
+        // TODO: Check that this cascades properly to rel tables
+        if ($this->_physioFileID == $physioFileID) {
+            $db->delete(
+                "physiological_task_event",
+                ['PhysiologicalTaskEventID' => $physiologicalTaskEventID]
+            );
+        }
+    }
+
+    /**
+     * Updates the event files associated with the given
+     * physiological file ID
+     *
+     * @return void
+     * @throws SodiumException
+     */
+    function updateFiles(): void
+    {
+        $db = \NDB_Factory::singleton()->database();
+
+        //Get data directory base path from Config
+        $dataDir = $db->pselectOne(
+            'SELECT Value
+            FROM Config AS config
+            INNER JOIN ConfigSettings AS c
+            ON c.Name=:name AND config.ConfigID=c.ID',
+            ['name' => 'dataDirBasepath']
+        );
+
+        $tsv = $db->pselect(
+            "SELECT
+        EventFileID   AS id,
+        FilePath      AS filePath,
+        ProjectID     AS projectID,
+        LastUpdate    AS lastUpdate,
+        LastWritten   AS lastWritten
+      FROM physiological_event_file
+      WHERE PhysiologicalFileID=:PFID
+      AND FileType='tsv'",
+            ['PFID' => $this->_physioFileID]
+        );
+
+        $tsvPath = count($tsv) > 0 ? $dataDir . $tsv[0]['filePath'] : '';
+
+        // Update files if files updated before database updated
+        if ($tsv[0]['lastWritten'] <= $tsv[0]['lastUpdate']) {
```

Comment:
We should check the validity of `$this->_physioFileID`. Maybe in constructor? 
If `$this->_physioFileID` does not exist in db, `$tsv` is empty, and I think this should trigger a warning here (`$tsv[0]` does not exist).

Reply:
Good find. The logic now addresses this corner case.